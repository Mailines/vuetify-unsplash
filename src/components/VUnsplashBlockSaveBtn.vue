<template>
  <v-app-bar app fixed bottom class="elevation-0" color="#00000000">
    <v-container fluid>
      <v-layout class="mx-8 py-1 d-flex justify-center">
        <v-btn
          class="my-1"
          x-large
          block
          :dark="isWhiteTextOnAccent && selectionSaveable"
          @click="saveSelection"
          :color="accentColor"
          :disabled="!selectionSaveable"
          :outlined="saveBtnOutlined"
          id="save-block-btn-v-unsplash"
          >{{ saveBtnLabel }}
          <v-icon class="ml-2">{{ saveBtnIcon }}</v-icon></v-btn
        >
      </v-layout>
    </v-container></v-app-bar
  >
</template>
<script>
export default {
  props: {
    accentColor: {
      type: String,
      required: true
    },
    isWhiteTextOnAccent: {
      type: Boolean,
      required: true
    },
    selectionSaveable: {
      type: Boolean,
      required: true
    },
    saveBtnLabel: {
      type: String,
      required: true
    },
    saveBtnOutlined: {
      type: Boolean,
      required: true
    },
    saveBtnIcon: {
      type: String,
      required: true
    }
  },
  methods: {
    saveSelection() {
      this.$emit("save");
    }
  }
};
</script>
