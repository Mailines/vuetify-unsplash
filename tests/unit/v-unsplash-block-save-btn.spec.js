import VUnsplashBlockSaveBtn from "../../src/components/VUnsplashBlockSaveBtn";
import { mount, createLocalVue } from "@vue/test-utils";
import Vuetify from "vuetify";
import Vue from "vue";

describe("VUnsplashBlockSaveBtn.vue", () => {
  let localVue;
  let wrapper;
  let vuetify;

  beforeEach(() => {
    localVue = createLocalVue();
    vuetify = new Vuetify();
    wrapper = mount(VUnsplashBlockSaveBtn, {
      localVue,
      vuetify,
      propsData: {
        accentColor: "#1976d2",
        isWhiteTextOnAccent: true,
        selectionSaveable: false,
        saveBtnLabel: "Save",
        saveBtnIcon: "mdi-content-save",
        saveBtnOutlined: true
      }
    });
  });
  it("renders and match snapshot", async () => {
    expect(wrapper.exists()).toBe(true);
    // With jest we can create snapshot files of the HTML output
    expect(wrapper.html()).toMatchSnapshot();
  });
  it("disabled the save button correctly", () => {
    expect(wrapper.find("#save-block-btn-v-unsplash").props("disabled")).toBe(
      true
    );
  });
  it("enables save button correctly when selectionSaveable=true", async () => {
    wrapper.setProps({
      selectionSaveable: true
    });
    await Vue.nextTick();
    expect(wrapper.find("#save-block-btn-v-unsplash").props("disabled")).toBe(
      false
    );
  });
  it("emits save event when save button is clicked", async () => {
    wrapper.setProps({
      selectionSaveable: true
    });
    await Vue.nextTick();
    wrapper.find("#save-block-btn-v-unsplash").trigger("click");
    expect(wrapper.emitted().save).toBeTruthy();
  });
});
