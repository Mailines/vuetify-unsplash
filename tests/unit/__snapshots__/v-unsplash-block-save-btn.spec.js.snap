// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VUnsplashBlockSaveBtn.vue renders and match snapshot 1`] = `
<header class="elevation-0 v-sheet theme--light v-toolbar v-toolbar--bottom v-app-bar v-app-bar--fixed" style="height: 64px; margin-top: 0px; transform: translateY(0px); left: 0px; right: 0px; background-color: rgba(0, 0, 0, 0); border-color: #00000000;">
  <div class="v-toolbar__content" style="height: 64px;">
    <div class="container container--fluid">
      <div class="layout mx-8 py-1 d-flex justify-center"><button type="button" disabled="disabled" class="my-1 v-btn v-btn--block v-btn--depressed v-btn--disabled v-btn--flat v-btn--outlined theme--light v-size--x-large" id="save-block-btn-v-unsplash"><span class="v-btn__content">Save
        <i aria-hidden="true" class="v-icon notranslate ml-2 mdi mdi-content-save theme--light"></i></span></button></div>
    </div>
  </div>
</header>
`;
